sklearn/ensemble/_stacking.py

```diff
@@ -0,0 +1,668 @@
+
+        **predict_params : dict of str -> obj
+            Parameters to the `predict` called by the `final_estimator`. Note
+            that this may be used to return uncertainties from some estimators
+            with `return_std` or `return_cov`. Be aware that it will only
+            accounts for uncertainty in the final estimator.
+
+        Returns
+        -------
+        y_pred : ndarray of shape (n_samples,) or (n_samples, n_output)
+            Predicted targets.
+        """
+        y_pred = super().predict(X, **predict_params)
+        return self._le.inverse_transform(y_pred)
+
+    @if_delegate_has_method(delegate='final_estimator_')
+    def predict_proba(self, X):
+        """Predict class probabilities for X using
+        `final_estimator_.predict_proba`.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        Returns
+        -------
+        probabilities : ndarray of shape (n_samples, n_classes) or \
+            list of ndarray of shape (n_output,)
+            The class probabilities of the input samples.
+        """
+        check_is_fitted(self)
+        return self.final_estimator_.predict_proba(self.transform(X))
+
+    def transform(self, X):
+        """Return class labels or probabilities for X for each estimator.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Training vectors, where `n_samples` is the number of samples and
+            `n_features` is the number of features.
+
+        Returns
+        -------
+        y_preds : ndarray of shape (n_samples, n_estimators) or \
+                (n_samples, n_classes * n_estimators)
+            Prediction outputs for each estimator.
+        """
+        return self._transform(X)
+
+
+class StackingRegressor(_BaseStacking, RegressorMixin):
+    """Stack of estimators with a final regressor.
+
+    Stacked generalization consists in stacking the output of individual
+    estimator and use a regressor to compute the final prediction. Stacking
+    allows to use the strength of each individual estimator by using their
+    output as input of a final estimator.
+
+    Note that `estimators_` are fitted on the full `X` while `final_estimator_`
+    is trained using cross-validated predictions of the base estimators using
+    `cross_val_predict`).
+
+    .. versionadded:: 0.22
+
+    Read more in the :ref:`User Guide <stacking>`.
+
+    Parameters
+    ----------
+    estimators : list of (str, estimator)
+        Base estimators which will be stacked together. Each element of the
+        list is defined as a tuple of string (i.e. name) and an estimator
+        instance. An estimator can be set to 'drop' using `set_params`.
+
+    final_estimator : estimator, default=None
```

Comment:
will it be more reasonable to set Ridge as default since our LogisticRegression is penalized by default? not sure

Reply:
We should not use `Ridge` but `RidgeCV` if we go for a penalized model.

Reply:
Why?

Reply:
From what I recall from discussions with @GaelVaroquaux, alpha in ridge will be more unstable and going to change greatly from dataset to dataset, having significant effect on the performance than the C of logistic regression.

Reply:
> From what I recall from discussions with @GaelVaroquaux, alpha in ridge will be more unstable and going to change greatly from dataset to dataset, having significant effect on the performance than the C of logistic regression.

If so, let's use LinearRegression. (are there any studies/ experiments?)

And is it possible to remove the default value of final_estimator?

Reply:
LinearRegression is a bad idea.

RidgeCV should work well. With LOO it is computationally cheap.

Reply:
I wonder if there is literature on using `stack_method_='predict_proba'` and a `final_estimator` equal to a pipeline with `NMF` as the first step.